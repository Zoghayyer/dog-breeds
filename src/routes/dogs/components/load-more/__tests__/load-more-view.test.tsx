import React from 'react';
import { shallow } from 'enzyme';
import LoadMoreView, { LoadMoreViewProps } from '../load-more-view';

describe('(View) LoadMoreView', () => {
  let component;
  const defaultProps: LoadMoreViewProps = {
    handleClick: () => ({}),
  };

  const shallowLoadMoreView = (extendedProps: Partial<LoadMoreViewProps> = {}) => {
    component = shallow(<LoadMoreView {...defaultProps} {...extendedProps} />);
  };

  beforeEach(() => shallowLoadMoreView());

  it('renders LoadMoreView component', () => {
    expect(component).toMatchSnapshot();
  });

  it('calls handleClick on button click', () => {
    const handleClickSpy = jest.fn();
    shallowLoadMoreView({ handleClick: handleClickSpy });

    component.find('.load-more-button').simulate('click');
    expect(handleClickSpy).toHaveBeenCalledTimes(1);
  });
});
