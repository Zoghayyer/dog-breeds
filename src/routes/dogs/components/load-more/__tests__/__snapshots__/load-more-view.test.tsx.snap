// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(View) LoadMoreView renders LoadMoreView component 1`] = `
<div
  className="d-flex justify-content-center mt-3"
>
  <Button
    className="load-more-button"
    onClick={[Function]}
    size="sm"
    theme="primary"
    type="button"
  >
    Load More
  </Button>
</div>
`;
